packages/phpstan-rules/tests/Rules/ForbiddenMethodCallByTypeInLocationRule/config/configured_rule.neon

```diff
@@ -0,0 +1,9 @@
+services:
+    -
+        class: Symplify\PHPStanRules\Rules\ForbiddenMethodCallByTypeInLocationRule
+        tags: [phpstan.rules.rule]
+        arguments:
+            forbiddenTypeInLocations:
+                Controller:
+                    - View\Helper
+                    - Symplify\PHPStanRules\Tests\Rules\ForbiddenMethodCallByTypeInLocationRule\Fixture\View\Helper\NumberHelper
```

Comment:
Seeing on 2nd change, this might be working slightly differently.
Instead of forbiding type in a type, this should be inversed - opt-in.

The configuration should allow to allow to use only in mentioned type. E.g.

- Only `*Repository` type can require a `Doctrine\ORM\EntityManager` type
- That means requiring `Doctrine\ORM\EntityManager` type anywhere else, e.g. in Controller, should fail

<br>

```yaml
 -
        class: Symplify\PHPStanRules\Rules\ExclusiveDependencyRule
        tags: [phpstan.rules.rule]
        arguments:
            allowedTypeByLocations:
                *Repository:
                    - Doctrine\ORM\EntityManager
```

<br>

```php
class SomeRepository
{
	public function __construct(\Doctrine\ORM\EntityManager $entityManager)
	{
	}
}
```

:heavy_check_mark: 

```php
class SomeController
{
	public function __construct(\Doctrine\ORM\EntityManager $entityManager)
	{
	}
}
```

:no_entry_sign: : 


Only the `__construct` method should be checked for this type.

What do you need me to add so you can be sure about the meaning of this rule? I'll try to narrow it down

Reply:
It is need Fnmatcher service then?

Reply:
@TomasVotruba do you want new rule named `ExclusiveDependencyRule` ?